### DISCUSSION TITLE
Is it possible to save a metamodel?

### DISCUSSION THREAD

**Discussion by wangluochao902** [textX/textX#340] [Q&A]
Hi, thanks for the useful tool. 

I have a large textx grammar file (about 5000 lines) and the `metamodel_from_str` function takes quite a long time (~0.6 second) in my case. Creating a metamodel every time is quite expensive. 

```
from textx import metamodel_from_str
import time
import pickle
start = time.time()
with open('grammer.tx', 'r') as f:
    s = f.read()
print('os time is', time.time() - start) # 0.005 second
meta = metamodel_from_str(s)
print('total time is', time.time()-start) # 0.6 second.
```

I tried to pickle the metamodel,
```
with open('meta.pkl', 'wb') as f:
    pickle.dump(meta, f)
```
 but got the following error:
```
TypeError: cannot serialize '_io.TextIOWrapper' object
```
Do you have recommendation on saving the metamodel object?

**Comment by igordejanovic** [comment]
textX currently doesn't support (meta)-model pickling.
Some related discussion regarding serialization can be found in #41, #285, #295

**Comment by krase** [comment]
I have a tool that parses files based on a textx grammar. It gets started many times using the same meta model every time.  
It would greatly improve the overall performance if the meta model would not need to be parsed/built on every run.  
So i also would appreciate to have the possibility to pickle the meta model.
